docs/src/content/guides/how-it-works.mdx

```diff
@@ -0,0 +1,120 @@
+# How Pigment CSS works
+
+Pigment CSS is a zero-runtime CSS-in-JS library. This means it does not have access to the end user's browser runtime which would be necessary to generate and insert authored CSS at runtime. Instead, it does all its processing at build time to pre-generate the CSS which then becomes part of the output bundle. This is why it cannot be consumed on its own. You must install the Pigment CSS plugin that corresponds to the framework you're using and configure your bundler accordingly (See the [integrations](/overview/integrations) page for more information).
+
+## Processor
+
+Pigment CSS uses the [WyW-in-JS](https://wyw-in-js.dev/) library that also powers [Linaria](https://linaria.dev/). It features a [processor](https://wyw-in-js.dev/how-to/custom-tagged-template#creating-a-processor) which makes it possible to create custom logic that's triggered by the presence of different imports from the library. The processor looks through the source code for `styled()`, `css()`, and other function calls and extracts the arguments to be evaluated. These values are then handed back to Pigment CSS for additional parsing and evaluation.
+
+For example, here's a simple usage of the `css()` function:
+
+```js
+// app.js
+import { css } from '@pigment-css/react-new';
+
+const testClass = css(({ theme }) => ({
+  padding: 0,
+  position: 'relative',
+  ...theme.typography.body1,
+}));
+```
+
+This snippet uses the `css()` call to generate and assign a class name string to the `testClass` variable.
+
+The corresponding (minimal) bundler configuration for this to work with the theme might look like:
+
+```ts title="bundler.config.ts"
+const customTheme = {
+  typography: {
+    body1: {
+      fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
+      fontWeight: 400,
+      fontSize: 16,
+      lineHeight: 1.5,
+      // ...others
+    },
+  },
+};
+
+const pigmentConfig = {
+  theme: customTheme,
+};
+```
+
+### 1. Detection and evaluation
+
+When the source file above (`app.js`) goes through the bundler transform, Pigment CSS's bundler plugin looks for the call site of `css` imported from `@pigment-css/react` and prepares an intermediate source file to get the actual values of the arguments of the `css()` call. This happens through WyW-in-JS's processors. In this case, the simplified intermediate code is:
+
+```js
+const _exp1 = ({ theme }) => ({
+  padding: 0,
+  position: 'relative',
+  ...theme.typography.body1,
+});
+
+module.exports = {
+  _exp1,
+};
+```
+
+The above code is then evaluated through the use of the `node:module` [module](https://nodejs.org/docs/v20.11.1/api/modules.html#module) which returns the evaluated values of the parameters.
```

Comment:
Do you mean https://nodejs.org/docs/v20.11.1/api/vm.html ?

Reply:
Also, I'd explain that the result of this step is that it obtains a static CSS fragment that is associated with the `css` call.

In the next step I'd say that now it generates a classname it can use in place of the `css()` call on js side, and use to wrap the styles with on CSS side